frameworks/Crystal/crystal/server.cr

```diff
@@ -14,9 +22,72 @@ server = HTTP::Server.new("0.0.0.0", 8080) do |context|
     response.status_code = 200
     response.headers["Content-Type"] = "text/plain"
     response.print "Hello, World!"
+  when "/db"
+    response.status_code = 200
+    response.headers["Content-Type"] = "application/json"
+    random_world.to_json(response)
+  when "/queries"
+    response.status_code = 200
+    response.headers["Content-Type"] = "application/json"
+
+    results = (1..sanitized_query_count(request)).map do
+      random_world
+    end
+
+    results.to_json(response)
+  when "/fortunes"
+    response.status_code = 200
+    response.headers["Content-Type"] = "text/html; charset=UTF-8"
+
+    data = fortunes
+    additional_fortune = {
+      id:      0,
+      message: "Additional fortune added at request time.",
+    }
+
+    data.push(additional_fortune)
+    data.sort_by! { |fortune| fortune[:message] }
```

Comment:
`sort_by!` is slower than using `sort! { |f1, f2| f1[:message] <=> f2[:message] }` because `sort_by!` tries to call the block only once per item (it allocates an entirely new array!). Since `f1[:message]` essentially has zero cost it's faster to do it like this.

@asterite This seems like a wart in the stdlib, thoughts?